From 44a750dd533e4364fc84ed52190687eb51f25a40 Mon Sep 17 00:00:00 2001
From: lyz <lyz@rock-chips.com>
Date: Wed, 1 Apr 2015 10:22:06 +0800
Subject: [PATCH] usb: dwc_otg: fix usb otg force_device_mode

TEST: set rockchip,usb-mode = <2> in dts and then the usb works

Tested-by: huang zhibao <hzb@rock-chips.com>
Signed-off-by: lyz <lyz@rock-chips.com>
---
 drivers/usb/dwc_otg_310/dwc_otg_pcd_linux.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/usb/dwc_otg_310/dwc_otg_pcd_linux.c b/drivers/usb/dwc_otg_310/dwc_otg_pcd_linux.c
index cc3afa20b6fe..99b3dd7c35ef 100755
--- a/drivers/usb/dwc_otg_310/dwc_otg_pcd_linux.c
+++ b/drivers/usb/dwc_otg_310/dwc_otg_pcd_linux.c
@@ -1752,6 +1752,12 @@ static void dwc_otg_pcd_work_init(dwc_otg_pcd_t *pcd,
 		pldata->dwc_otg_uart_mode(pldata, PHY_USB_MODE);
 	}
 	schedule_delayed_work(&pcd->check_id_work, 8 * HZ);
+	if (otg_dev->core_if->usb_mode == USB_MODE_FORCE_DEVICE) {
+		pcd->vbus_status = 0;
+		dwc_otg_core_init(otg_dev->core_if);
+		cil_pcd_start(otg_dev->core_if);
+		dwc_otg_pcd_start_check_vbus_work(pcd);
+	}
 }
 
 #endif /* DWC_HOST_ONLY */
-- 
2.35.3

